### PR TITLE
Support more languages

### LINKED ISSUE #32: [feature] Support more languages?
Such as the list: <https://github.com/nvim-treesitter/nvim-treesitter#supported-languages>

### PR DESCRIPTION
**Pull request by Freed-Wu**
Fix #32

### CODE REVIEW

**Comment by grantjenks** [comment]
Look at the past merges. You have to change more than repos.txt

**Comment by Freed-Wu** [comment]
Update build.py

**Comment by grantjenks** [comment]
Also the readme and the tests.

**Comment by Freed-Wu** [comment]
Updated

**Comment by Freed-Wu** [comment]
Related discussion: <https://github.com/nvim-treesitter/nvim-treesitter/discussions/5588>

**Comment by Freed-Wu** [comment]
@grantjenks

**Comment by grantjenks** [comment]
You can’t just delete all the licenses from the readme. Distributing in binary form doesn’t necessarily remove all license restrictions.

**Comment by Freed-Wu** [comment]
Can we have some method to update license automatically like <https://github.com/nvim-treesitter/nvim-treesitter/discussions/5588#discussioncomment-7415038>?

**Comment by Freed-Wu** [comment]
> You can’t just delete all the licenses from the readme.

I have added the licenses.

> Distributing in binary form doesn’t necessarily remove all license restrictions.

[nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) doesn't add any license in README. Is it necessary?

**Comment by Freed-Wu** [comment]
@grantjenks

**Comment by Freed-Wu** [comment]
And, I advise to add a scheduled CI/CD to update the version.

**Comment by Freed-Wu** [comment]
@grantjenks Any other problem? If you are busy to review this PR, can you add me to collaborators?

**Comment by grantjenks** [comment]
Looking at the project licenses, I don't think you can add GPL'd code and maintain this project as Apache 2.

ChatGPT thinks so too: https://chat.openai.com/share/fe181dca-6ebc-4965-890e-d8ef20cfcb6b

IANAL

**Comment by Freed-Wu** [comment]
> add GPL'd code and maintain this project as Apache 2.

Where is GPL's code? This project's license shouldn't be changed.

File: README.rst [hunk 1]
```diff
@@ -191,147 +191,233 @@ under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
...
-
-* https://github.com/stadelmanma/tree-sitter-fixed-form-fortran - licensed under
-  the MIT License.
-
 .. _`tree-sitter`: https://tree-sitter.github.io/
+
+* https://git.sr.ht/~rockorager/tree-sitter-scfg — licensed under the MIT License
+* https://git.sr.ht/~yotam/tree-sitter-smali — licensed under the MIT License
+* https://gitlab.com/WhyNotHugo/tree-sitter-jsonc.git — licensed under the MIT License
+* https://gitlab.com/gabmus/tree-sitter-blueprint.git — licensed under the GNU General Public License v3.0
```
Comment (grantjenks) [inline README.rst:202]: GPL'd code

File: README.rst [hunk 2]
```diff
@@ -191,147 +191,233 @@ under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
...
+* https://github.com/ObserverOfTime/tree-sitter-ssh-config — licensed under the MIT License
+* https://github.com/ObserverOfTime/tree-sitter-xml — licensed under the MIT License
+* https://github.com/PRQL/tree-sitter-prql — licensed under the MIT License
+* https://github.com/Philipp-M/tree-sitter-ungrammar — licensed under the MIT License
+* https://github.com/Philipp-M/tree-sitter-yuck — licensed under the MIT License
+* https://github.com/PorterAtGoogle/tree-sitter-textproto — licensed under the ISC License
+* https://github.com/PrestonKnopp/tree-sitter-gdscript — licensed under the MIT License
+* https://github.com/PrestonKnopp/tree-sitter-godot-resource — licensed under the MIT License
+* https://github.com/RaafatTurki/tree-sitter-sxhkdrc — licensed under the MIT License
+* https://github.com/RubixDev/ebnf — licensed under the GNU General Public License v3.0
```
Comment (grantjenks) [inline README.rst:249]: GPL'd code

**Review by grantjenks — CHANGES_REQUESTED** [review]
There will need to be an ALLOWLIST of Apache2 compatible software licenses.

GPLv3 is not compatible with Apache2.

**Comment by sansmoraxz** [comment]
Any updates on this?

**Comment by sansmoraxz** [comment]
Could you move the hashes of the repo to stable commit (or the commit with latest release attached if possible). I don't think all of them are stable.

**Comment by sansmoraxz** [comment]
Also a lot of them are missing `src/parser.c`. (Written in JS)

Those will fail to build.

**Comment by Freed-Wu** [comment]
> a lot of them are missing src/parser.c

Not sure why missing. Let us delete them before we know why.

**Comment by Freed-Wu** [comment]
> There will need to be an ALLOWLIST of Apache2 compatible software licenses.

How to get the allowlist?

**Comment by Freed-Wu** [comment]
There are many languages information is stored in <https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lua/nvim-treesitter/parsers.lua#L79-L2269> and updated very frequently.
We should have a method to extract them to get the latest parsers.

**Comment by sansmoraxz** [comment]
Is it really a better approach though?

I find locking down to commit id's to be a much more sensible option.

**Pull request closed without merging**